app/models/certificate.rb

```diff
@@ -1,16 +1,21 @@
 class Certificate < ApplicationRecord
-  after_initialize :build_values
-
   belongs_to :volunteer
   belongs_to :user, -> { with_deleted }
 
   def build_values
     return unless text_body.nil?
-    update(text_body: default_text_body, function: DEFAULT_FUNCTION,
-      hours: volunteer.hours.total_hours, minutes: volunteer.hours.minutes_rest,
-      assignment_kinds: volunteer.assignment_kinds, volunteer_contact: convert_volunteer_contact,
-      institution: DEFAULT_INSTITUTION)
-    update(volunteer.assignments_duration)
+    self.text_body ||= default_text_body
+    self.function ||= DEFAULT_FUNCTION
+    self.hours ||= volunteer.hours.total_hours
+    self.minutes ||= volunteer.hours.minutes_rest
+    self.assignment_kinds ||= volunteer.assignment_kinds
+    self.volunteer_contact ||= convert_volunteer_contact
+    self.institution ||= DEFAULT_INSTITUTION
+    self.group_offer ||= has_group_offer
```

Comment:
this would fail if `self.group_offset` is false, and it doesn't make sense to initialize it with a `nil` value